Fix layout tests in simulator

pvollan opened the pull request:
#### 47d87cd99fd28559352bec7d539b9a2d0a984121
<pre>
Fix layout tests in simulator
<a href="https://rdar.apple.com/121324669">rdar://121324669</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=267822">https://bugs.webkit.org/show_bug.cgi?id=267822</a>

Reviewed by Brent Fulgham.

Fix an issue where the Simulator entitlements are not actually applied. Also use the same bundle ID
for WebKit extensions on Simulator as on device. Additionally, the test runner app needs an extra
entitlement for launching extensions to succeed. This patch changes the bundle IDs for the XPC
services on Simulator where extension based launch is enabled. Otherwise, these bundle IDs will
conflict with the extension bundle IDs.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUExtension.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkingExtension.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/WebContentCaptivePortalExtension.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
* Source/WebKit/Configurations/WebContentExtension.xcconfig:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::createNewWebProcess):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:

Canonical link: <a href="https://commits.webkit.org/273374@main">https://commits.webkit.org/273374@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e240df3a58169267c0333dfa23108c8743185f85

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35232 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14166 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37359 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37988 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31795 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36395 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16547 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11233 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30688 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35778 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11982 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31410 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10503 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/35472 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10558 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31518 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39234 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32044 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31849 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36542 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10690 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8617 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34555 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12458 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11213 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4548 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11518 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
```diff
@@ -148,19 +148,26 @@ static void launchWithExtensionKit(ProcessLauncher& processLauncher, ProcessLaun
 
 void ProcessLauncher::launchProcess()
 {
+    //WTF::sleep(30_s);
```
Nit - it seems like this was just for debugging?
```suggestion
```

whsieh replied:
(It also wasn't clear if the `fprintf` and `fflush` below were intended — if so, it might be to make `"Returning early"` a bit more descriptive)

pvollan replied:
Ah, good points! I'll remove these in the next patch.

Thanks for reviewing!

brentfulgham commented on Source/WebKit/Scripts/process-entitlements.sh:
```diff
@@ -495,7 +495,7 @@ plistbuddy Clear dict
 if [[ "${WK_PLATFORM_NAME}" =~ .*simulator ]]
 then
-    if [[ "${PRODUCT_NAME}" != adattributiond && "${PRODUCT_NAME}" != webpushd ]]; then
+    if [[ -e "${CODE_SIGN_ENTITLEMENTS}" && "${PRODUCT_NAME}" != adattributiond && "${PRODUCT_NAME}" != webpushd ]]; then
```
Ah, good change.

brentfulgham reviewed (approved):
Looks good. r=me

brentfulgham commented on Source/WebKit/Configurations/WebContentService.xcconfig:
```diff
@@ -30,6 +30,13 @@ PRODUCT_NAME_Development = com.apple.WebKit.WebContent.Development;
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphonesimulator17.1*] = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphonesimulator17.2*] = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphonesimulator17.3*] = $(PRODUCT_NAME);
+PRODUCT_BUNDLE_IDENTIFIER[sdk=iphonesimulator18.0*] = $(PRODUCT_NAME);
```
This pattern doesn't seem very maintainable. We should think about how to make this more automatic.

pvollan replied:
That is a good point! We may be able to remove most of this as we move forward.

Thanks for reviewing!

brentfulgham commented on Source/WebKit/Shared/AuxiliaryProcessExtensions/AuxiliaryProcessExtension.entitlements:
```diff
@@ -6,7 +6,7 @@
 	<key>com.apple.private.webkit.use-xpc-endpoint</key>
 	<true/>
-	<key>om.apple.private.extensionkit.host-requirement-exemption</key>
+	<key>com.apple.private.extensionkit.host-requirement-exemption</key>
```
Oops!

brentfulgham reviewed (approved):
Looks good. r=me

pvollan committed:
Fix layout tests in simulator
rdar://121324669
https://bugs.webkit.org/show_bug.cgi?id=267822

Reviewed by Brent Fulgham.

Fix an issue where the Simulator entitlements are not actually applied. Also use the same bundle ID
for WebKit extensions on Simulator as on device. Additionally, the test runner app needs an extra
entitlement for launching extensions to succeed. This patch changes the bundle IDs for the XPC
services on Simulator where extension based launch is enabled. Otherwise, these bundle IDs will
conflict with the extension bundle IDs.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/Configurations/BaseXPCService.xcconfig:
* Source/WebKit/Configurations/GPUExtension.xcconfig:
* Source/WebKit/Configurations/GPUService.xcconfig:
* Source/WebKit/Configurations/NetworkingExtension.xcconfig:
* Source/WebKit/Configurations/NetworkService.xcconfig:
* Source/WebKit/Configurations/WebContentCaptivePortalExtension.xcconfig:
* Source/WebKit/Configurations/WebContentService.xcconfig:
* Source/WebKit/Configurations/WebContentExtension.xcconfig:
* Source/WebKit/Scripts/process-entitlements.sh:
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::createNewWebProcess):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:

Canonical link: https://commits.webkit.org/273374@main
